### PR TITLE
Los go

### PR DESCRIPTION
**Pull request by lospro7**

### CODE REVIEW

File: config/default.conf.json [hunk 1]
```diff
@@ -3,14 +3,20 @@
...
+            "StoreRoot" : "",
+            "Region" : "us-east-1",
+            "NamePathRegex" : "",
+            "NamePathMap" : "${ImageName}/${ImageSize}"
         },
-        "Local" : {
-            "storeRoot": "/Users/jarvis/imagestore"
+        {
+            "Type" : "local",
+            "StoreRoot": "/Users/jarvis/imagestore",
```
Comment (briankassouf) [inline config/default.conf.json]: jarvis

File: factory.go [hunk 2]
```diff
@@ -2,22 +2,54 @@ package main
 
 import (
 	"github.com/gophergala/ImgurGo/imagestore"
+	"github.com/mitchellh/goamz/aws"
+	"github.com/mitchellh/goamz/s3"
```
Comment (briankassouf) [inline factory.go]: Why does the factory need to know about external s3 classes?

**Comment by briankassouf** [comment]
Woot :bomb:

File: factory.go (hunk 2 above)
Comment (lospro7) [inline factory.go]: the s3 dependency gets instantiated here

**Comment by Jasdev** [comment]
:+1:

**Pull request merged**